packages/rum-core/src/domain/assembly.ts

```diff
@@ -157,15 +159,13 @@ function needToAssembleWithAction(
   return [RumEventType.ERROR, RumEventType.RESOURCE, RumEventType.LONG_TASK].indexOf(event.type) !== -1
 }
 
-function getSessionType() {
-  return navigator.userAgent.indexOf('DatadogSynthetics') === -1 && !getSyntheticsContext()
-    ? SessionType.USER
-    : SessionType.SYNTHETICS
-}
+export const SYNTHETICS_TEST_ID_COOKIE_NAME = 'datadog-synthetics-public-id'
+export const SYNTHETICS_RESULT_ID_COOKIE_NAME = 'datadog-synthetics-result-id'
 
 function getSyntheticsContext() {
-  const testId = (window as BrowserWindow)._DATADOG_SYNTHETICS_PUBLIC_ID
-  const resultId = (window as BrowserWindow)._DATADOG_SYNTHETICS_RESULT_ID
+  const testId = (window as BrowserWindow)._DATADOG_SYNTHETICS_PUBLIC_ID || getCookie(SYNTHETICS_TEST_ID_COOKIE_NAME)
+  const resultId =
+    (window as BrowserWindow)._DATADOG_SYNTHETICS_RESULT_ID || getCookie(SYNTHETICS_RESULT_ID_COOKIE_NAME)
 
   if (typeof testId === 'string' && typeof resultId === 'string') {
```

Comment:
Out of curiosity, what is this test for? 
Is there any previous issue related to it?

Reply:
Mmh this test is a bit too preventive, indeed. I don't recall any issue where values where truthy without being a string. But I guess it's fine nonetheless?

Reply:
yes